server/src/item.rs

```diff
@@ -115,62 +123,95 @@ impl Item {
             )));
         };
 
-        let mut inner = self.inner.lock().await;
-
-        match session.aes_key() {
-            Some(key) => {
-                let decrypted = oo7::crypto::decrypt(secret, &key, &iv).map_err(|err| {
-                    custom_service_error(&format!("Failed to decrypt secret {err}."))
-                })?;
-                inner.set_secret(decrypted);
+        match &mut *self.inner.lock().await {
+            InnerItem::Locked(_inner) => {
+                tracing::error!("Cannot set secret of a locked object `{}`", self.path);
+                Err(ServiceError::IsLocked(format!(
+                    "Cannot set secret of a locked object `{}`.",
+                    self.path
+                )))
             }
-            None => {
-                inner.set_secret(secret);
+            InnerItem::Unlocked(inner) => {
+                match session.aes_key() {
+                    Some(key) => {
+                        let decrypted = oo7::crypto::decrypt(secret, &key, &iv).map_err(|err| {
+                            custom_service_error(&format!("Failed to decrypt secret {err}."))
+                        })?;
+                        inner.set_secret(decrypted);
+                    }
+                    None => {
+                        inner.set_secret(secret);
+                    }
+                }
+
+                Ok(())
             }
         }
-
-        Ok(())
     }
 
     #[zbus(property, name = "Locked")]
     pub async fn is_locked(&self) -> bool {
-        self.locked.load(std::sync::atomic::Ordering::Relaxed)
+        matches!(&*self.inner.lock().await, InnerItem::Locked(_))
     }
 
     #[zbus(property, name = "Attributes")]
     pub async fn attributes(&self) -> HashMap<String, String> {
-        self.inner
-            .lock()
-            .await
-            .attributes()
-            .iter()
-            .map(|(k, v)| (k.to_owned(), v.to_string()))
-            .collect()
+        match &*self.inner.lock().await {
+            InnerItem::Locked(_inner) => todo!(),
```

Comment:
I assume this is because we can't return custom errors from properties, can we return zbus::fdo errors? if so we can map the error for now. Otherwise maybe return Default::default() would be enough with a todo message pointing to the corresponding zbus issue.